weldx/core.py

```diff
@@ -132,7 +132,11 @@ def equals(
                 equality = simplify(self.expression - other.expression) == 0
 
             if check_parameters:
-                equality = equality and self._parameters == other.parameters
+                from weldx.util import compare_nested
```

Comment:
do we have some ring dependencies on top level import here?

Reply:
Can't tell for this particular import, but I had some ring dependency issues. Maybe we should start transferring the code to have one file per class like in the transformations package.

Reply:
unless it is indeed necessary here I would prefer to have these kind of imports at the file scope for consistency

Reply:
After a short test, this causes ring dependency issues.

Reply:
we should take a look into this @marscher @vhirtham soon